backend/src/main/java/com/llm_service/llm_service/controller/conversation/ConversationController.java

```diff
@@ -39,10 +36,21 @@ public class ConversationController {
             })
     @Operation(summary = "Get all conversations")
     @GetMapping(value = "/api/v1/paid/conversation")
-    public ResponseEntity<List<ConversationResponseCompact>> getAllConversations() throws UnauthorizedException {
-        return ResponseEntity.ok(conversationService.getAll().stream()
+    public ResponseEntity<List<ConversationResponseCompact>> getAllConversations(
+            @RequestParam(defaultValue = "0") int page, @RequestParam(defaultValue = "10") int pageSize)
+            throws UnauthorizedException {
+
+        List<Conversation> allConversations = conversationService.getAll();
```

Comment:
the pagination should be on the level of the repository so the Hibernate will not fetch more data if it is not needed.